### PR TITLE
Reworked CI and cleaned up other parts

### PR DESCRIPTION
**Pull request by TrueBrain**
This was long overdue.

In summary:
- Don't deploy to AWS anymore; we haven't used that in a long time now.
- Use a better way to deploy to Cloudflare.
- Move out as much of the GitHub workflows as we could to a single repository, so we have a single place to maintain things from.
- Update Python dependencies

And, maybe most importantly:
- When you merge things go to production; no longer any need to tag.
- This is possible because PRs now come with a preview. Well, after this PR is merged, that is.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,7 +1,3 @@
 [flake8]
 max-line-length = 120
 inline-quotes = double
-
-# We use 'black' for coding style; these next two warnings are not PEP-8
-# compliant.
-ignore = E203, W503
```
Comment (LordAro) [inline .flake8:7]: why is this not relevant anymore? Even if it's added automatically by the actions, it'll confuse things when running flake8 locally, no?

Comment (TrueBrain) [reply .flake8:7]: I no longer get any of these warnings on repositories. So I guess this is fixed in flake8 or black :)

It isn't automatically added; also locally you no longer need it.

Comment (LordAro) [reply .flake8:7]: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#flake8 Would suggest that E203 is still necessary, at least

Comment (TrueBrain) [reply .flake8:7]: Good call! Fixed :)

**Review by LordAro — APPROVED** [review]

**Pull request merged**